force-app/main/default/lwc/geGift/geSoftCredits.js

```diff
@@ -35,6 +38,9 @@ class SoftCredits {
     }
 
     addProcessedSoftCredits(processedSoftCredits) {
+        if (typeof processedSoftCredits === 'string') {
+            processedSoftCredits = JSON.parse(processedSoftCredits);
```

Comment:
I think we should try to avoid reassignment in this spot and in the constructor too. 

`this._processedSoftCredits = parseIfString(processedSoftCredits);` possibly?